src/workers/python/PythonWorker.worker.ts

```diff
@@ -78,8 +78,10 @@ class PythonWorker extends Backend {
     override async autocomplete(context: WorkerAutocompleteContext):
         Promise<CompletionResult | null> {
         // Do not await as not strictly required to compute autocompletions
-        this.pyodide.loadPackagesFromImports(context.text);
-        const result = this.convert(await this.papyros.autocomplete(context));
+        const result = (await Promise.all([
+            this.pyodide.pyimport("pyodide_worker_runner").install_imports(context.text),
+            this.convert(await this.papyros.autocomplete(context))
```

Comment:
If you await this here then it's no longer a promise and the Promise.all is redundant.

Reply:
Indeed, the await in line 83 is in the wrong location.